src/Catel.Core/Catel.Core.Shared/IoC/ServiceLocator.cs

```diff
@@ -656,6 +658,74 @@ public void RemoveAllTypes(Type serviceType)
         #endregion
 
         #region Methods
+        /// <summary>
+        /// Determines whether the specified service type is registered as open generic.
+        /// </summary>
+        /// <param name="serviceType">The type of the service.</param>
+        /// <param name="tag">The tag to register the service with. The default value is <c>null</c>.</param>
+        /// <returns><c>true</c> if the specified service type is registered; otherwise, <c>false</c>.</returns>
+        private bool IsTypeRegisteredAsOpenGeneric(Type serviceType, object tag = null)
+        {
+            if (serviceType.IsGenericTypeEx())
```

Comment:
Now this has been migrated to a new method, we can simplify it (if (!serviceType.IsGenerictypeEx) return